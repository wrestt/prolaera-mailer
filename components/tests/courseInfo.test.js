import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/button';
import CourseInfo from '../../components/course/courseInfo';

Enzyme.configure({ adapter: new Adapter() });

const course = {
  by: 'John Doe',
  course_id: '1234-1234-1234-1234',
  courseAudience: 'Accountants',
  delivery_method: 1,
  hours: [
    {
      credits: 1,
      subject_area: 'Tax Preparation'
    }
  ],
  level: 'Basic',
  name: 'Tax Preparation 102',
  objectives: '<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>',
  prep: '<p>Read course introduction</p>',
  prerequisites: '<p>Tax Prep 101</p>',
  summary: '<p>This course is the second installation of Tax Prep focused on general tax planning.</p>',
  header: 'Your course Tax Preparation 101 is ready for review',
  footer: 'Log in below'
};
const headerText = "You've been assigned a new course, " + `${course.name}`;
const buttonText = 'Log In';

describe('CourseInfo component custom input tests', () => {
  it('creates and checks a snapshot of CourseInfo html', async () => {
    let courseHtml = renderer.create(<CourseInfo {...course} />);
    let courseJson = courseHtml.toJSON();
    expect(courseJson).toMatchSnapshot();
  });

  it('checks all custom inputs', async () => {
    const wrapper = shallow(<CourseInfo {...course} headerProps={headerText} buttonProps={buttonText} />);
    expect(wrapper.contains(<p>You've been assigned a new course, Tax Preparation 102</p>)).toBe(true);
    expect(
      wrapper.contains(
        <h3>
          Target Audience: <span style={{ fontWeight: 'normal' }}>Accountants</span>
        </h3>
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <h3>
          Delivery Method: <span style={{ fontWeight: 'normal' }}>Group-Live</span>
        </h3>
      )
    ).toBe(true);
    expect(
      wrapper.contains(<li style={{ fontSize: '18px', fontWeight: 'normal' }}>Tax Preparation - 1 Hour(s)</li>)
    ).toBe(true);
    expect(
      wrapper.contains(
        <h3>
          Level: <span style={{ fontWeight: 'normal' }}>Basic</span>
        </h3>
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <h3>
          Learning Objectives:
          <div
            style={{ fontSize: '18px', fontWeight: 'normal' }}
            dangerouslySetInnerHTML={{
              __html: '<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>'
            }}
          />
        </h3>
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          style={{ fontSize: '18px', fontWeight: 'normal' }}
          dangerouslySetInnerHTML={{ __html: '<p>Read course introduction</p>' }}
        />
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          style={{ fontSize: '18px', fontWeight: 'normal' }}
          dangerouslySetInnerHTML={{ __html: '<p>Tax Prep 101</p>' }}
        />
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          style={{ fontSize: '18px', fontWeight: 'normal' }}
          dangerouslySetInnerHTML={{
            __html: '<p>This course is the second installation of Tax Prep focused on general tax planning.</p>'
          }}
        />
      )
    ).toBe(true);
    expect(wrapper.contains(<Button text={'Log In'} />)).toBe(true);
  });
});
