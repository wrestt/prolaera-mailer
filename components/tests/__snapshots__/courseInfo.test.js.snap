// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseInfo component custom input tests creates and checks a snapshot of CourseInfo html 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "borderRadius": "5px",
      "boxShadow": "1px 1px 8px darkgray",
      "display": "block",
      "fontFamily": "Arial, sans-serif",
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "500px",
      "padding": "20px",
      "textAlign": "center",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    className="courseReview"
    valign="top"
    width="100%"
  >
    <tbody>
      <tr>
        <td
          align="left"
        >
          <div
            style={
              Object {
                "fontSize": "18px",
                "fontWeight": "normal",
                "marginTop": "5px",
              }
            }
          >
            <em>
              <h2>
                Tax Preparation 102
              </h2>
            </em>
          </div>
        </td>
      </tr>
      <tr>
        <td
          align="left"
        >
          <div
            style={
              Object {
                "paddingLeft": "40px",
              }
            }
          >
            <h3>
              By: 
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                John Doe
              </span>
            </h3>
            <h3>
              Recommended CPE Credit(s):
            </h3>
            <ul>
              <li
                style={
                  Object {
                    "fontSize": "18px",
                    "fontWeight": "normal",
                  }
                }
              >
                Tax Preparation
                 - 
                1
                 Hour(s)
              </li>
            </ul>
            <h3>
              Delivery Method: 
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Group-Live
              </span>
            </h3>
            <h3>
              Price: 
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Free
              </span>
            </h3>
            <h3>
              Target Audience: 
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Accountants
              </span>
            </h3>
            <h3>
              Prep:
            </h3>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Read course introduction</p>",
                }
              }
              style={
                Object {
                  "fontSize": "18px",
                  "fontWeight": "normal",
                }
              }
            />
            <h3>
              Prerequisites:
            </h3>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Tax Prep 101</p>",
                }
              }
              style={
                Object {
                  "fontSize": "18px",
                  "fontWeight": "normal",
                }
              }
            />
            <h3>
              Level: 
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Basic
              </span>
            </h3>
            <h3>
              Learning Objectives:
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>",
                  }
                }
                style={
                  Object {
                    "fontSize": "18px",
                    "fontWeight": "normal",
                  }
                }
              />
            </h3>
            <h3>
              Description:
            </h3>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>This course is the second installation of Tax Prep focused on general tax planning.</p>",
                }
              }
              style={
                Object {
                  "fontSize": "18px",
                  "fontWeight": "normal",
                }
              }
            />
          </div>
        </td>
      </tr>
      <tr>
        <td
          align="left"
        >
          <div>
            <span
              className="buttonSpan"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
